src/common/http-signature/verify-signature.middleware.ts

```diff
@@ -2,24 +2,27 @@ import { Injectable, NestMiddleware } from '@nestjs/common';
 import { NextFunction, Request, Response } from 'express';
 import { LTO } from '@ltonetwork/lto';
 import { verify } from '@ltonetwork/http-message-signatures';
+import { ConfigService } from '../config/config.service';
 
 @Injectable()
 export class VerifySignatureMiddleware implements NestMiddleware {
   private readonly lto: LTO;
 
-  constructor() {
-    this.lto = new LTO();
+  constructor(private readonly config: ConfigService) {
+    const networkID = this.config.getNetworkId();
+    this.lto = new LTO(networkID);
```

Comment:
Creating an LTO object doesn't work because it assumes that you know the network id in advance. Use a callback instead.

Alternatively, you may limit the service to only mainnet or testnet, but then this should be done throughout the app.